language-server/src/main/kotlin/tools/samt/ls/SamtTextDocumentService.kt

```diff
@@ -118,13 +115,12 @@ class SamtTextDocumentService(private val workspaces: Map<URI, SamtWorkspace>) :
     override fun semanticTokensFull(params: SemanticTokensParams): CompletableFuture<SemanticTokens> =
         CompletableFuture.supplyAsync {
             val path = params.textDocument.uri.toPathUri()
-            val workspace = getWorkspace(path)
 
-            val fileInfo = workspace?.get(path) ?: return@supplyAsync SemanticTokens(emptyList())
+            val fileInfo = workspace.getFile(path) ?: return@supplyAsync SemanticTokens(emptyList())
 
             val tokens: List<Token> = fileInfo.tokens
             val fileNode: FileNode = fileInfo.fileNode ?: return@supplyAsync SemanticTokens(emptyList())
-            val globalPackage: Package = workspace.samtPackage ?: return@supplyAsync SemanticTokens(emptyList())
+            val globalPackage: Package = workspace.getFolderSnapshot(path)?.globalPackage ?: return@supplyAsync SemanticTokens(emptyList())
```

Comment:
You could use `getRootPackage` here
```suggestion
            val globalPackage: Package = workspace.getRootPackage(path) ?: return@supplyAsync SemanticTokens(emptyList())
```